analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/InsecureDeserialization.cs

```diff
@@ -56,66 +53,63 @@ public InsecureDeserialization(IAnalyzerConfiguration analyzerConfiguration)
         }
 
         protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(VisitDeclaration, SyntaxKind.ClassDeclaration, SyntaxKindEx.RecordClassDeclaration);
-
-        private void VisitDeclaration(SyntaxNodeAnalysisContext context)
-        {
-            if (!IsEnabled(context.Options) || context.ContainingSymbol.Kind != SymbolKind.NamedType)
-            {
-                return;
-            }
-
-            var declaration = (TypeDeclarationSyntax)context.Node;
-            if (!HasConstructorsWithParameters(declaration))
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
             {
-                // If there are no constructors, or if these don't have parameters, there is no validation done
-                // and the type is considered safe.
-                return;
-            }
+                var declaration = (TypeDeclarationSyntax)c.Node;
+                if (!IsEnabled(c.Options)
```

Comment:
If we revert this, and the next `if`, we can merge them to a nice positive condition (except the redundant record).

I'd put the `IsRedundantPositionalRecordContet()` first. It will not be called often and it's cheaper than `IsEnabled`